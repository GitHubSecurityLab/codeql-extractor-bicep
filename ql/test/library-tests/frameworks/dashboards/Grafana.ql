private import bicep

query predicate grafana(Dashboards::GrafanaResource grafana) { any() }

query predicate publicGrafana(Dashboards::PublicGrafanaResource publicGrafana) { any() }

query predicate grafanaProperties(
  Dashboards::GrafanaResource grafana, Dashboards::GrafanaProperties::Properties properties
) {
  grafana.getProperties() = properties
}

query predicate grafanaEnterpriseConfig(
  Dashboards::GrafanaResource grafana,
  Dashboards::GrafanaProperties::EnterpriseConfigurations enterpriseConfig
) {
  grafana.getProperties().getEnterpriseConfigurations() = enterpriseConfig
}
