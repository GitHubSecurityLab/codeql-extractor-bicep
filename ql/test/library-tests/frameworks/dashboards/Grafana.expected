grafana
| app.bicep:5:1:12:1 | GrafanaResource |
| app.bicep:15:1:24:1 | GrafanaResource |
| app.bicep:15:1:24:1 | Public Grafana Resource |
| app.bicep:27:1:36:1 | GrafanaResource |
| app.bicep:39:1:49:1 | GrafanaResource |
| app.bicep:52:1:65:1 | GrafanaResource |
| app.bicep:68:1:92:1 | GrafanaResource |
| app.bicep:68:1:92:1 | Public Grafana Resource |
| app.bicep:95:1:107:1 | GrafanaResource |
| app.bicep:110:1:123:1 | GrafanaResource |
| app.bicep:126:1:146:1 | GrafanaResource |
| app.bicep:149:1:162:1 | GrafanaResource |
| app.bicep:165:1:178:1 | GrafanaResource |
| app.bicep:181:1:195:1 | GrafanaResource |
| app.bicep:198:1:216:1 | GrafanaResource |
| app.bicep:219:1:231:1 | GrafanaResource |
| app.bicep:234:1:305:1 | GrafanaResource |
| app.bicep:234:1:305:1 | Public Grafana Resource |
| app.bicep:308:1:317:1 | GrafanaResource |
publicGrafana
| app.bicep:15:1:24:1 | GrafanaResource |
| app.bicep:15:1:24:1 | Public Grafana Resource |
| app.bicep:68:1:92:1 | GrafanaResource |
| app.bicep:68:1:92:1 | Public Grafana Resource |
| app.bicep:234:1:305:1 | GrafanaResource |
| app.bicep:234:1:305:1 | Public Grafana Resource |
grafanaProperties
| app.bicep:5:1:12:1 | GrafanaResource | app.bicep:8:15:8:16 | GrafanaProperties |
| app.bicep:15:1:24:1 | GrafanaResource | app.bicep:18:15:20:3 | GrafanaProperties |
| app.bicep:15:1:24:1 | Public Grafana Resource | app.bicep:18:15:20:3 | GrafanaProperties |
| app.bicep:27:1:36:1 | GrafanaResource | app.bicep:30:15:32:3 | GrafanaProperties |
| app.bicep:39:1:49:1 | GrafanaResource | app.bicep:45:15:45:16 | GrafanaProperties |
| app.bicep:52:1:65:1 | GrafanaResource | app.bicep:61:15:61:16 | GrafanaProperties |
| app.bicep:68:1:92:1 | GrafanaResource | app.bicep:77:15:84:3 | GrafanaProperties |
| app.bicep:68:1:92:1 | Public Grafana Resource | app.bicep:77:15:84:3 | GrafanaProperties |
| app.bicep:95:1:107:1 | GrafanaResource | app.bicep:98:15:103:3 | GrafanaProperties |
| app.bicep:110:1:123:1 | GrafanaResource | app.bicep:113:15:119:3 | GrafanaProperties |
| app.bicep:126:1:146:1 | GrafanaResource | app.bicep:129:15:142:3 | GrafanaProperties |
| app.bicep:149:1:162:1 | GrafanaResource | app.bicep:152:15:158:3 | GrafanaProperties |
| app.bicep:165:1:178:1 | GrafanaResource | app.bicep:168:15:174:3 | GrafanaProperties |
| app.bicep:181:1:195:1 | GrafanaResource | app.bicep:184:15:191:3 | GrafanaProperties |
| app.bicep:198:1:216:1 | GrafanaResource | app.bicep:201:15:212:3 | GrafanaProperties |
| app.bicep:219:1:231:1 | GrafanaResource | app.bicep:222:15:227:3 | GrafanaProperties |
| app.bicep:234:1:305:1 | GrafanaResource | app.bicep:243:15:295:3 | GrafanaProperties |
| app.bicep:234:1:305:1 | Public Grafana Resource | app.bicep:243:15:295:3 | GrafanaProperties |
| app.bicep:308:1:317:1 | GrafanaResource | app.bicep:311:15:313:3 | GrafanaProperties |
grafanaEnterpriseConfig
| app.bicep:95:1:107:1 | GrafanaResource | app.bicep:99:31:102:5 | EnterpriseConfigurations |
| app.bicep:234:1:305:1 | GrafanaResource | app.bicep:250:31:253:5 | EnterpriseConfigurations |
| app.bicep:234:1:305:1 | Public Grafana Resource | app.bicep:250:31:253:5 | EnterpriseConfigurations |
